hudi-client/src/main/java/org/apache/hudi/client/utils/SparkEngineUtils.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.utils;
+
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.apache.spark.api.java.function.Function;
+
+import java.util.List;
+
+/**
+ * Util class for Spark Engine.
+ */
+public class SparkEngineUtils {
```

Comment:
Actually, If we depend on this class, we can not call it **spark free**.

Reply:
@yanghua IMO, this is the same Purpose of [HUDI-678](https://github.com/apache/incubator-hudi/pull/1418), just make this class spark-free. After all spark RDD calculate move to this class , we can abstract the FlinkEngine like this.

Reply:
Also I would have a base interface `EngineContext` (let's not please overload Utils anymore.. I am trying to move us towards SRP principles).. and subclass `SparkRDDEngineContext` (we may add a DataFrame engine, Flink Engine).. and generify the code such that we pass the engineContext once to HoodieWriteClient and the rest of the code can execute.. 

This sore of PoC will be extremely valuable to use at this stage, than doing classes one by one.. We will take a long time to be done :) .. @yanghua would you also agree with my thoughts here

Reply:
Yes, I suggest whether we can temporarily block this PR. We internally expect that an incomplete version based on Flink implementation will be given this Friday. Can we look at its implementation then discuss furthermore?

Reply:
yes.. I am with you... @yanghua and @hddong

Reply:
Yes, agree too @yanghua @vinothchandar